actgen/agents/action_dqn.py

```diff
@@ -44,7 +44,10 @@ def _get_q_values_for_state(self, x):
         return self.q(qnet_input)
 
     def _make_qnet(self, n_features, n_actions, params):
+        use_dropout = 'regularization' in params and 'dropout' in params['regularization']
         return MLP(n_inputs=n_features+n_actions,
                    n_outputs=1,
                    n_hidden_layers=params['n_hidden_layers'],
-                   n_units_per_layer=params['n_units_per_layer'])
+                   n_units_per_layer=params['n_units_per_layer'],
+                   use_dropout=use_dropout,
+                   dropout_rate=params['dropout_rate'])
```

Comment:
`use_dropout` has a check to make sure it's in the params, but `dropout_rate` doesn't have a check. Should it have one? Do we really need both hyperparameters? What if we used dropout_rate = 0 for no dropout?

Reply:
ah, good suggestion I like that! 

In case that `dropout_rate = 0`, we still need to handle this as a special case when creating the network, instead of always using a dropout layer with `torch.nn.Dropout(dropout_rate)` right? Because `torch.nn.Dropout(0)` might create inefficiencies?

Reply:
If the default dropout value is `False`, then we can simply do `if dropout`, because when `dropout=0` or `0.0` the conditional will evaluate to False

Reply:
We may also need to turn off dropout during `change_q`